kroxylicious-runtime/src/main/java/io/kroxylicious/proxy/KafkaProxy.java

```diff
@@ -207,6 +208,9 @@ public void shutdown() throws InterruptedException {
             STARTUP_SHUTDOWN_LOGGER.info("Shutting down");
             endpointRegistry.shutdown().handle((u, t) -> {
                 bindingOperationProcessor.close();
+                if (filterChainFactory != null) {
+                    filterChainFactory.close();
```

Comment:
Should we wrap this in a try catch/finally to ensure we progress to `serverEvenGroup.shutdownGracefully`

Reply:
I ended up just moving it after the gracefulshutdown of the event loops, because that way we should have the guarantee that nothing is going to call `createFilter()` after the factory has been shutdown, avoiding some nasty `ISE` in the logs.